src/Check/RabbitMq/AmqpExt/RabbitMqQueueCheck.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FiveLab\Component\Diagnostic\Check\RabbitMq\AmqpExt;
+
+use FiveLab\Component\Diagnostic\Check\RabbitMq\RabbitMqConnectionParameters;
+use FiveLab\Component\Diagnostic\Result\Failure;
+use FiveLab\Component\Diagnostic\Result\ResultInterface;
+use FiveLab\Component\Diagnostic\Result\Success;
+use FiveLab\Component\Diagnostic\Result\Warning;
+
+/**
+ * RabbitMQ queue specific checks
+ */
+class RabbitMqQueueCheck extends AbstractRabbitMqCheck
+{
+    /**
+     * @var string
+     */
+    private $queueName;
+
+    /**
+     * @var int|null
+     */
+    private $maxMessages;
+
+    /**
+     * must be between 0 and 100
+     * @var int
+     */
+    private $maxWarningPercentage;
+
+    /**
+     * @var int|null
+     */
+    private $minMessages;
+
+    /**
+     * @param RabbitMqConnectionParameters $connectionParameters
+     * @param string                       $queueName
+     */
+    public function __construct(RabbitMqConnectionParameters $connectionParameters, string $queueName)
+    {
+        parent::__construct($connectionParameters);
+
+        $this->queueName = $queueName;
+    }
+
+    /**
+     * @param int $maxMessages
+     */
+    public function setMaxMessages(int $maxMessages): void
```

Comment:
All parameters must passed via `__constructor` and all `health check service` must be immutable.